test/uk/gov/hmrc/cgtpropertydisposals/connectors/returns/SubmitReturnsConnectorSpec.scala

```diff
@@ -56,25 +63,49 @@ class SubmitReturnsConnectorSpec extends WordSpec with Matchers with MockFactory
     )
   )
 
-  val submitReturnRequest: SubmitReturnRequest = {
-    sample[SubmitReturnRequest].copy(completeReturn = sample[CompleteReturn]
-      .copy(triageAnswers = sample[CompleteSingleDisposalTriageAnswers].copy(numberOfProperties = One))
-    )
-  }
+
 
   val connector = new SubmitReturnsConnectorImpl(mockHttp, new ServicesConfig(config, new RunMode(config, Mode.Test)))
 
   "SubmitReturnsConnectorImpl" when {
 
     implicit val hc: HeaderCarrier = HeaderCarrier()
-    val expectedHeaders            = Map("Authorization" -> s"Bearer $desBearerToken", "Environment" -> desEnvironment)
+    val expectedHeaders = Map("Authorization" -> s"Bearer $desBearerToken")//, "Environment" -> desEnvironment)
 
     def expectedSubmitReturnUrl(cgtReference: String) =
       s"""http://localhost:7022/capital-gains-tax/cgt-reference/$cgtReference/return"""
 
     "handling request to submit return" must {
 
+      "do a post http call and pass correct parameters" in {
+        import Version7._
+        val schemaToBeValidated = Json.fromJson[SchemaType](
+          Json.parse(Source.fromInputStream(this.getClass.getResourceAsStream("/schema.json")).mkString)
+        ).get
+
+        val validator = SchemaValidator(Some(Version7))
+
+        for( a <- 1 to 1000){
+          val submitReturnRequest: SubmitReturnRequest =
+            sample[SubmitReturnRequest].copy(completeReturn = sample[CompleteReturn]
+              .copy(propertyAddress = sample[UkAddress], triageAnswers = sample[CompleteSingleDisposalTriageAnswers].copy(numberOfProperties = One)))
+          val ppdReturnDetails = PPDReturnDetails(submitReturnRequest)
+          val desSubmitReturnRequest = DesSubmitReturnRequest(ppdReturnDetails)
+          val json = Json.toJson(desSubmitReturnRequest)
+          val validationResult = validator.validate(schemaToBeValidated, json)
+
+          withClue(s"****** Validating json ******" + json + "****** with validation result ******" + validationResult) {
+            validationResult.isSuccess shouldBe true
```

Comment:
is there any way to make the test print out a helpful message if it fails, e.g. something like
```scala
if(validationResult.isSuccess) ()
else fail(s"validation failed: ${validationResult.description}")
```
not sure what methods there are on `validationResult` which might help.